homeassistant/components/forked_daapd/media_player.py

```diff
@@ -667,70 +671,106 @@ async def async_play_media(
-                    + self._tts_pause_time
-                )
-            except asyncio.TimeoutError:
-                self._tts_requested = False
-                _LOGGER.warning("TTS request timed out")
-            self._tts_playing_event.clear()
-            # TTS done, return to normal
-            await self.async_turn_on()  # restore outputs and volumes
-            if saved_mute:  # mute if we were muted
-                await self.async_mute_volume(True)
-            if self._use_pipe_control():  # resume pipe
-                await self._api.add_to_queue(
-                    uris=self._sources_uris[self._source], clear=True
-                )
-                if saved_state == MediaPlayerState.PLAYING:
-                    await self.async_media_play()
-            else:  # restore stashed queue
-                if saved_queue:
-                    uris = ""
-                    for item in saved_queue["items"]:
-                        uris += item["uri"] + ","
-                    await self._api.add_to_queue(
-                        uris=uris,
-                        playback="start",
-                        playback_from_position=saved_queue_position,
-                        clear=True,
-                    )
-                    await self._api.seek(position_ms=saved_song_position)
-                    if saved_state == MediaPlayerState.PAUSED:
-                        await self.async_media_pause()
-                    elif saved_state != MediaPlayerState.PLAYING:
-                        await self.async_media_stop()
-        else:
+        elif media_type not in CAN_PLAY_TYPE:
             _LOGGER.debug("Media type '%s' not supported", media_type)
+            return
+
+        if kwargs.get(ATTR_MEDIA_ANNOUNCE):
+            return await self._async_announce(media_id)
+
+        if (
+            enqueue := kwargs.get(ATTR_MEDIA_ENQUEUE, MediaPlayerEnqueue.REPLACE)
+        ) is True:
+            enqueue = MediaPlayerEnqueue.ADD
+        if enqueue in {MediaPlayerEnqueue.ADD, MediaPlayerEnqueue.REPLACE}:
+            return await self._api.add_to_queue(
+                uris=media_id,
+                playback="start",
+                clear=enqueue == MediaPlayerEnqueue.REPLACE,
+            )
+
+        current_position = next(
+            (
+                item["position"]
+                for item in self._queue["items"]
+                if item["id"] == self._player["item_id"]
+            ),
+            0,
+        )
+        if enqueue == MediaPlayerEnqueue.NEXT:
+            return await self._api.add_to_queue(
+                uris=media_id,
+                playback="start",
+                position=current_position + 1,
+            )
+        # enqueue == MediaPlayerEnqueue.PLAY
+        return await self._api.add_to_queue(
+            uris=media_id,
+            playback="start",
+            position=current_position,
+            playback_from_position=current_position,
+        )
```

Comment:
The old behavior was to just play everything as a TTS. That behavior is now limited to when the service call comes in as ATTR_MEDIA_ANNOUNCE. Control is passed to the `_async_announce()` method below.
If the service call does not have ATTR_MEDIA_ANNOUNCE, it should have ATTR_MEDIA_ENQUEUE (if it has neither we assume the default of REPLACE (see my comment here: https://github.com/home-assistant/architecture/issues/765#issuecomment-1242928411).
In that same post, it says if ATTR_MEDIA_ENQUEUE is True, we assume ADD.
See https://developers.home-assistant.io/docs/core/entity/media-player?_highlight=media&_highlight=player#play-media for the service call details.